int16[] gridX
int16[] gridY
int16[] height
int16[] width
float32[] theta
int8 number_of_path
